### PR TITLE
#77 smarti widget invalidation

### PR DESCRIPTION
**Pull request by mrsimpson**
This PR adds a button to the configuration of the application which allows to reload Smarti frontend from the Smarti server.
Furthermore, the buffered state is getting refreshed on a daily basis.

### CODE REVIEW

File: packages/dbs-ai/server/methods/getSmartiUiScript.js [hunk 1]
```diff
@@ -1,36 +1,58 @@
...
+	const response = HTTP.get(`${ DBS_AI_SMARTI_URL }plugin/v1/rocket.chat.js`);
+	if (response.statusCode === 200) {
+		script = response.content;
+
+		if (!script) {
+			SystemLogger.error('Could not extract script from Smarti response');
 			throw new Meteor.Error('no-smarti-ui-script');
+		} else {
+			// add pseudo comment in order to make the script appear in the frontend as a file. This makes it de-buggable
+			script = `${ script } //# sourceURL=SmartiWidget.js`;
```
Comment (mrsimpson) [inline packages/dbs-ai/server/methods/getSmartiUiScript.js]: This suffix after the script body makes a resource named `SmartiWidget.js` appear in the IDE
Webstorm:
smarti widget webstorm

Chrome:
debug smarti

**Comment by ruKurz** [comment]
Loading the widget-script via inline `<script>` tag is not the very best option. It lacks in capabilities for debugging (see the added JS-comment at the end of the Smarti widget file) and caching (having a button to flush the server side puffer and a 24h timeout). Both extensions are just workarounds. The correct solution for this would be loading the script file via `src` attribute.

**Comment by ruKurz** [comment]
Tested, changes work on my machine.

**Comment by ruKurz** [comment]
Unfortunately the user does not get any feedback when hitting the button. So it feels like nothing happen at all.

**Review by ruKurz — COMMENTED** [review]
**Tested**
Tested, changes work on my machine.


**Workaround**
Loading the widget-script via inline <script> tag is not the very best option. It lacks in capabilities for debugging (see the added JS-comment at the end of the Smarti widget file) and caching (having a button to flush the server side puffer and a 24h timeout). Both extensions are just workarounds. The correct solution for this would be loading the script file via src attribute.

Loading the JS-source-file via src attribute requires a direct connection from the client/browser to the Smarti application. Since Smarti (knowledge base) is not designed to be reachable over the internet Rocket.Chat must load the script server side and expose it via HTTP again.

For this reason both of the extensions included in this PR are required anyway.


**User Feedback**
Unfortunately the user does not get any feedback when hitting the button. So it feels like nothing happen at all.

**Comment by mrsimpson** [comment]
@ruKurz added error handling and UI propagation.

success
error

**Pull request merged**